UI/src/view/point_asset/trafficLightForm.js

```diff
@@ -0,0 +1,339 @@
+(function(root) {
+    root.TrafficLightForm = function() {
+        PointAssetForm.call(this);
+        var me = this;
+
+        var propertyOrdering = [
+            'trafficLight_type',
+            'trafficLight_info',
+            'trafficLight_municipality_id',
+            'trafficLight_structure',
+            'trafficLight_height',
+            'trafficLight_sound_signal',
+            'trafficLight_vehicle_detection',
+            'trafficLight_push_button',
+            'trafficLight_relative_position',
+            'location_coordinates_x',
+            'location_coordinates_y',
+            'trafficLight_lane',
+            'trafficLight_lane_type',
+            'trafficLight_state',
+            'suggest_box',
+            'counter'
+        ];
+
+        //might be needed if the client wants to see properties of the old traffic light form
+       /* var oldAssetPropertyOrdering = [
+            'suggest_box',
+            'counter'
+        ];*/
+
+        var renderOldToNewTrafficLightForm = function (selectedAsset, authorizationPolicy) {
+            var wrapper = $('<div class="wrapper">');
+            var formRootElement = $('<div class="form form-horizontal form-dark form-point-asset">');
+            var oldToNewButton =  $('' +
+            '    <div class="form-group point-asset edit-only">' +
+            '      <label class="control-label"></label>' +
+            '      <button id="old-to-new-traffic-light" class="btn btn-secondary old-to-new-asset">Siirry muokkaamaan liikennevaloa</button>' +
+            '    </div>');
+
+            formRootElement = formRootElement
+                .append(oldToNewButton);
+                //might be needed if the client wants to see properties of the old traffic light form
+                //.append(me.renderComponents(selectedAsset, oldAssetPropertyOrdering, authorizationPolicy));
+
+            return wrapper.append(formRootElement);
+        };
+
+        var isOld = function(asset){
+            return _.head(me.getProperties(asset.propertyData, 'trafficLight_type').values).propertyValue === "";
```

Comment:
Suggestion:
```js
return _.isEmpty( _.head(me.getProperties(asset.propertyData, 'trafficLight_type').values).propertyValue )
```

Reply:
Applied

Reply:
Although this looks like a reasonable alteration it cannot be implemented like this. Because _.isEmpty does not support numbers and a new traffic light will have **values: [ {propertyValue: 1} ]** so it will return true for the verification above instead of false. It will work for old format traffic lights but not for new ones.